code/modules/integrated_electronics/subtypes/input.dm

```diff
@@ -339,6 +352,85 @@
 		set_pin_data(IC_OUTPUT, 7, tr)
 		set_pin_data(IC_OUTPUT, 8, H.density)
 		set_pin_data(IC_OUTPUT, 9, H.opacity)
+		set_pin_data(IC_OUTPUT, 10, get_turf(H))
+		push_data()
+		activate_pin(2)
+	else
+		activate_pin(3)
+
+/obj/item/integrated_circuit/input/turfpoint
+	name = "Tile pointer"
+	desc = "This circuit will get tile ref with given relative coordinates."
+	extended_desc = "If the machine	cannot see the target, it will not be able to calculate the correct direction.\
+	This circuit works only inside an assembly."
+	icon_state = "numberpad"
+	complexity = 5
+	inputs = list("X" = IC_PINTYPE_NUMBER,"Y" = IC_PINTYPE_NUMBER)
+	outputs = list("tile" = IC_PINTYPE_REF)
+	activators = list("calculate dir" = IC_PINTYPE_PULSE_IN, "on calculated" = IC_PINTYPE_PULSE_OUT,"not calculated" = IC_PINTYPE_PULSE_OUT)
+	spawn_flags = IC_SPAWN_RESEARCH
+	power_draw_per_use = 40
+
+/obj/item/integrated_circuit/input/turfpoint/do_work()
+	if(!assembly)
+		activate_pin(3)
+		return
+	var/turf/T = get_turf(assembly)
+	var/target_x = CLAMP(T.x + get_pin_data(IC_INPUT, 1), 0, world.maxx)
+	var/target_y = CLAMP(T.y + get_pin_data(IC_INPUT, 2), 0, world.maxy)
+	var/turf/A = locate(target_x, target_y, T.z)
+	set_pin_data(IC_OUTPUT, 1, null)
+	if(!A || !(A in view(T)))
+		activate_pin(3)
+		return
+	else
+		set_pin_data(IC_OUTPUT, 1, WEAKREF(A))
+	push_data()
+	activate_pin(2)
+
+/obj/item/integrated_circuit/input/turfscan
+	name = "tile analyzer"
+	desc = "This circuit can analyze the contents of the tile containing the referenced object, and can read letters on floor."
+	icon_state = "video_camera"
+	complexity = 5
+	inputs = list(
+		"target" = IC_PINTYPE_REF
+		)
+	outputs = list(
+		"located ref" 		= IC_PINTYPE_LIST,
+		"Written letters" 	= IC_PINTYPE_STRING,
+		"area"				= IC_PINTYPE_STRING
+		)
+	activators = list(
+		"scan" = IC_PINTYPE_PULSE_IN,
+		"on scanned" = IC_PINTYPE_PULSE_OUT,
+		"not scanned" = IC_PINTYPE_PULSE_OUT
+		)
+	spawn_flags = IC_SPAWN_RESEARCH
+	power_draw_per_use = 40
+	cooldown_per_use = 10
+
+/obj/item/integrated_circuit/input/turfscan/do_work()
+	var/atom/H = get_pin_data_as_type(IC_INPUT, 1, /atom)
+	var/turf/T = get_turf(src)
+	var/turf/E = get_turf(H)
+	var/AN = get_area_name(E)
+	if(!istype(H)) //Invalid input
+		return
+
+	if(H in view(T)) // This is a camera. It can't examine thngs,that it can't see.
```

Comment:
H is the atom, T is the turf the atom is on. H is literally in the contents of T (or in the contents of something that is), there should not be a scenario where they do not see eachother?

Reply:
T is turf, where assembly is.

Reply:
This is actually arsserpentarium's code that I updated, but yes, H is the scanned object, and T is the turf of the circuit/assembly itself.

Reply:
oh right, you're correct